Source/VisualStudioTools/Private/BlueprintReferencesCommandlet.cpp

```diff
@@ -0,0 +1,220 @@
+	{
+		PackageFilePath = FPaths::ConvertRelativePathToFull(MoveTemp(PackageFile));
+	}
+	else
+	{
+		PackageFilePath = "[Invalid file path]";
+	}
+
+	Json->WriteObjectStart();
+	Json->WriteValue(TEXT("name"), BlueprintClassName);
+	Json->WriteValue(TEXT("path"), PackageFilePath);
+	Json->WriteObjectEnd();
+}
+
+static void SerializeBlueprints(
+	TSharedRef<JsonWriter>& Json, const TMap<FString, FAssetData>& InAssets)
+{
+	Json->WriteIdentifierPrefix(TEXT("blueprints"));
+	Json->WriteArrayStart();
+
+	for (auto& Item : InAssets)
+	{
+		const FString& BlueprintClassName = Item.Key;
+		const FAssetData& Asset = Item.Value;
+		SerializeBlueprintReference(Json, BlueprintClassName, Asset);
+	}
+
+	Json->WriteArrayEnd();
+}
+
+static void SerializeMetadata(
+	TSharedRef<JsonWriter>& Json, int TotalAssetCount)
+{
+	Json->WriteIdentifierPrefix(TEXT("metadata"));
+	Json->WriteObjectStart();
+	{
+		Json->WriteValue(TEXT("asset_count"), TotalAssetCount);
+	}
+	Json->WriteObjectEnd();
+}
+
+static void SerializeResults(
+	const TMap<FString, FAssetData>& InAssets,
+	FArchive& OutArchive,
+	int TotalAssetCount)
+{
+	TSharedRef<JsonWriter> Json = JsonWriter::Create(&OutArchive);
+	Json->WriteObjectStart();
+
+	SerializeBlueprints(Json, InAssets);
+	SerializeMetadata(Json, TotalAssetCount);
+
+	Json->WriteObjectEnd();
+	Json->Close();
+}
+} // namespace VisualStudioTools
+
+static constexpr auto SymbolParamVal = TEXT("symbol");
+
+UVsBlueprintReferencesCommandlet::UVsBlueprintReferencesCommandlet()
+	: Super()
+{
+	HelpDescription = TEXT("Commandlet for generating data used by Blueprint support in Visual Studio.");
+
+	HelpParamNames.Add(SymbolParamVal);
+	HelpParamDescriptions.Add(TEXT("[Optional] Fully qualified symbol to search for in the blueprints."));
+
+	HelpUsage = TEXT("<Editor-Cmd.exe> <path_to_uproject> -run=VsBlueprintReferences -output=<path_to_output_file> -symbol=<ClassName::FunctionName> [-unattended -noshadercompile -nosound -nullrhi -nocpuprofilertrace -nocrashreports -nosplash]");
+}
+
+int32 UVsBlueprintReferencesCommandlet::Run(
+	TArray<FString>& Tokens,
+	TArray<FString>& Switches,
+	TMap<FString, FString>& ParamVals,
+	FArchive& OutArchive)
+{
+	GIsRunning = true; // Required for the blueprint search to work.
+
+	FString* ReferencesSymbol = ParamVals.Find(SymbolParamVal);
+	if (ReferencesSymbol->IsEmpty())
+	{
+		UE_LOG(LogVisualStudioTools, Error, TEXT("Missing required symbol parameter."));
+		PrintHelp();
+		return -1;
+	}
+
+	FString Function;
+	FString ClassNameNative;
+	if (!ReferencesSymbol->Split(TEXT("::"), &ClassNameNative, &Function))
+	{
+		UE_LOG(LogVisualStudioTools, Error, TEXT("Reference parameter should be in the qualified 'NativeClassName::MethodName' format."));
+		PrintHelp();
+		return -1;
+	}
+
+	FString ClassName = VisualStudioTools::StripClassPrefix(std::move(ClassNameNative));
+	FString SearchValue = FString::Printf(TEXT("Nodes(\"Native Name\"=+%s & ClassName=K2Node_CallFunction)"), *Function, *ClassName);
```

Comment:
Is the `*Function` supposed to be here? There is just a single `%`

Reply:
Yes, the *Function is what is intended, but the *ClassName should not, it's a left over from a previous version of the qeury. That is different from the `ClassName` field in the query, which is constant for the K2Node.

I initially have this here as a way to scope the results further, but it causes the query to not work for static methods. Those need to match nodes that do not have any class information in the FindInBlueprints metadata.